shepherd_bms/src/analyzer.cpp

```diff
@@ -177,6 +178,34 @@ void Analyzer::calcContDCL()
     }
 }
 
+void Analyzer::calcCCL()
+{
+    int16_t currentLimit = 0x7FFF;
+
+    for(uint8_t c = 0; c < NUM_CHIPS; c++)
+    {
+        for(uint8_t cell = 0; cell < NUM_CELLS_PER_CHIP; cell++)
+        {   
+            // Apply equation
+            uint16_t tmpCCL = (MAX_VOLT - ((bmsdata->chipData[c].openCellVoltage[cell] + VOLT_SAG_MARGIN) * 10000)) / (bmsdata->chipData[c].cellResistance[cell] * 10);
+            // Multiplying resistance by 10 to convert from mOhm to Ohm and then to Ohm * 10000 to account for the voltage units
+
+            //Taking the minimum DCL of all the cells
+            if(tmpCCL < currentLimit) currentLimit = tmpCCL;
+        }
+    }
+
+    // ceiling for current limit
+    if (currentLimit > MAX_CELL_CURR) 
+    {
+        bmsdata->dischargeLimit = MAX_CELL_CURR;
+    }
```

Comment:
we would not want to be changing the dischargeLimit in this calc. Instead, i think bmsdata should have a "chargeLimit" value that you can swap in